src/resource.rs

```diff
@@ -139,6 +144,72 @@ impl Debug for LowerFunc {
     }
 }
 
+/// Sealed trait for shared type parametrised functionality between [`TypeDef`] and [`OpDef`].
+pub trait TypeParametrisedInternal: sealed::SealedDef {
```

Comment:
This is not `Internal` - it's `pub` ! (Maybe you should merge this into `SealedDef` tho I think it's more useful to keep than `CustomConcrete`). The use case for parameterising over types `<T: TypeParametrisedInternal>` is *small* but I admit non-zero.

Also, the correct spelling is `parameterised`